### PR TITLE
Us 42

### LINKED ISSUE #41: User Story 42, Admin disables a merchant account
```
As an admin merchant
When I visit the merchant index page
And I click on a "disable" button for an enabled merchant
I am returned to the admin's merchant index page
And I see a flash message that the merchant's account is now disabled
And I see that the merchant's account is now disabled
This merchant cannot log in
```

DO THESE STORIES FIRST:
- User Story 41, Admin visits Merchant Index Page

### Checklist:

- [x] base tests are written
- [x] base code is written to pass tests
- [x] edge case ("sad path") tests written (if required)
- [x] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #227

child of #274

depends on #224

### PR DESCRIPTION
**Pull request by WHomer**
What does this PR do? 
* adds logic to not allow inactive users to login, across all roles. 
* adds structure to the flash messaging.
* resolves #41 & #40

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -85,10 +85,14 @@
     </nav>
 
     <% flash.each do |message| %>
-    <%= sanitize(message[1]) %>
+      <div id="flash-message" class="alert alert-primary alert-dismissible fade show" role="alert">
+        <%= sanitize(message[1]) %>
+        <button type="button" class="close" data-dismiss="alert" aria-label="Close">
+          <span aria-hidden="true">&times;</span>
+        </button>
+      </div>
```
Comment (BrennanAyers) [inline app/views/layouts/application.html.erb:93]: I like it!

File: app/views/merchants/index.html.erb [hunk 2]
```diff
@@ -14,9 +14,9 @@
           <li class="list-group-item text-center">Location: <%= merchant.city %>, <%= merchant.state %></li>
           <li class="list-group-item text-center">Joined: <%= merchant.created_at.strftime("%B %d, %Y") %></li>
           <% if current_admin? && merchant.active? %>
-            <li class="list-group-item text-center"><%= link_to "Disable", root_path %></li>
+            <li class="list-group-item text-center"><%= link_to "Disable", "/merchant/disable/#{merchant.id}", :method=> :patch %></li>
```
Comment (BrennanAyers) [inline app/views/merchants/index.html.erb:17]: This seems janky, but I also can't think of a cleaner way of doing this.

File: config/routes.rb [hunk 3]
```diff
@@ -28,6 +28,8 @@
 
   get '/dashboard', to: 'merchants#show'
   get '/merchants', to: 'merchants#index' # a link to see all merchants ("/merchants") -- User Story 2, Visitor Navigation
+  patch 'merchant/enable/:id', to: 'merchants#enable'
+  patch 'merchant/disable/:id', to: 'merchants#disable'
```
Comment (BrennanAyers) [inline config/routes.rb:32]: Yeah...

File: spec/features/merchants/index_spec.rb [hunk 4]
```diff
@@ -80,8 +80,44 @@
...
+        expect(current_path).to eq(merchants_path)
+
+        within("#flash-message") do
+          expect(page).to have_content("Merchant's account is now enabled")
+        end
+
+        within "#merchant-#{@m2.id}" do
+          expect(page).to have_link("Disable")
+        end
+      end
```
Comment (BrennanAyers) [inline spec/features/merchants/index_spec.rb:120]: Should we have expectations like (@m1.active).to be true/false? @WHomer

**Review by BrennanAyers — APPROVED** [review]

File: spec/features/merchants/index_spec.rb (hunk 4 above)
Comment (WHomer) [reply spec/features/merchants/index_spec.rb:120]: good point. I added this.
